src/base/emu-i386/simx86/trees.c

```diff
@@ -1258,6 +1258,20 @@ int InvalidateNodeRange(int al, int len, unsigned char *eip)
 	    */
 	    /* Exclude last instruction, as there is no need to break
 	     * node after last instruction (it ends there anyway). */
+	    if (!eip && P0 != (unsigned)-1) {
```

Comment:
Can we have different funcs,
rather that checking everything
for weird values?

Reply:
Or for example old_func_that_takes_eip(FindEIP(P0)).